desc/coils.py

```diff
@@ -930,6 +930,58 @@ def body(B, x):
             B = rpz2xyz_vec(B, x=coords[:, 0], y=coords[:, 1])
         return B
 
+    def compute_minimum_distance(
+        self,
+        params=None,
+        grid=None,
+        transforms=None,
+        **kwargs,
+    ):
+        """Compute minimum distance between coils in the coilset.
+
+        Parameters
+        ----------
+        params : dict or array-like of dict, optional
+            parameters to pass to curves, either the same for all curves,
+            or one for each member
+        grid : Grid, int or None or array-like, optional
+            Grid used to discretize coil, the same for all coils. If an integer, uses
+            that many equally spaced points.
+
+        Returns
+        -------
+        d_min : float
+            the minimum coil-coil distance between coils in the coilset.
+        """
+        if params is None:
+            params = [get_params(["x_s", "x", "s", "ds"], coil) for coil in self]
+            for par, coil in zip(params, self):
+                par["current"] = coil.current
+
+        coil_coords = self.compute(
+            "x",
+            grid=grid,
+            params=params,
+            basis="xyz",
+            transforms=transforms,
+            **kwargs,
+        )
+        # set some large value for initial separation
+        minSeparation2 = 1e20
+        for whichCoil1 in range(len(self)):
```

Comment:
the double loop here will be really slow to compile, probably better to just stack the output data and do a vectorized operation. Assuming theres O(10) coils with O(100) pts per coil, thats only a 1000x1000 matrix which is fine. Also makes it easier to get min distance per coil and do softmin etc.